modules/javafx.controls/src/main/java/javafx/scene/control/ListView.java

```diff
@@ -1306,6 +1306,14 @@ public ListViewBitSetSelectionModel(final ListView<T> listView) {
                 while (c.next()) {
                     final T selectedItem = getSelectedItem();
                     final int selectedIndex = getSelectedIndex();
+                    int addedItemOffset = -1;
+
+                    if (c.wasAdded()
+                            && selectedItem != null
+                            && (addedItemOffset = c.getAddedSubList().indexOf(selectedItem)) >= 0
+                            && selectedIndex == c.getFrom() + addedItemOffset) {
+                        doSelectionUpdate = false;
```

Comment:
which scenario does this code path handle outside of tests?

I could never hit a breakpoint in line 1315, except in ensureEditorValueDoesNotChangeWhenCurrentEditorValueIsAddedToItemsList().

Reply:
It's only when `ComboBox.getEditor().setText(...)` is called, which doesn't seem to immediately update the current value. This shouldn't really happen in most applications.